### ISSUE TITLE
Cannot read property 'swaggerOptions' of undefined, part 2

### ISSUE THREAD

**Issue by rf-** [swagger-api/swagger-ui#1944]
I left this as a comment on #560, but @webron suggested opening a new issue.

Basically the problem is that Swagger UI includes its own copy of Backbone with [a monkey patch](https://github.com/swagger-api/swagger-ui/blob/master/lib/backbone-min.js#L5-L15) appended to it. This causes hard-to-debug problems for people who want to embed Swagger UI into an app that has its own copy of Backbone already.

I would love it if Swagger UI were available in a single-file, self-contained build made with Webpack or something, but short of doing that, it would be good if the Backbone views that need this behavior could just override their own constructors instead of depending on a monkey patch.

**Comment by rf-** [comment]
Sorry, I realized my wording may not be clear: I'm talking about problems caused by not using Swagger UI's copy of Backbone, but instead using a normal copy of Backbone obtained from somewhere else.

**Comment by rf-** [comment]
Bump?

**Comment by fehguy** [comment]
A new version of Swagger-UI will remove this dependency.  Until then, we don't have a choice but to rely on our patched version.

**Comment by rf-** [comment]
Couldn't you just update the constructors of the views that are relying on the patch to copy the options themselves?

**Comment by rf-** [comment]
Anyway that's cool about the updated version, looking forward to it. Thanks for the response.

**Comment by mpospelov** [comment]
👍

**Comment by rsmithlal** [comment]
I needed to modify this to get it working with my app that already uses backbone. Instead of the monkeypatch of the stock `Backbone.View`, why didn't you extend it and monkeypatch the extended version and use that for Swagger-UI? That way you don't break our apps. Anyway, that's what I did in my instance. I extended Backbone.View and applied the monkeypatch to the extended version. Then, I replaced all instances of `Backbone.View` in your code with `Backbone.SwaggerView`.

``` js
Backbone.SwaggerView = Backbone.View.extend();

Backbone.SwaggerView = (function(View) {
   return View.extend({
        constructor: function(options) {
            this.options = options || {};
            View.apply(this, arguments);
        }
    });
})(Backbone.SwaggerView);
```
